### DISCUSSION TITLE
[@typescript/jsrsasign] /modules/KJUR/crypto/ECDSA.d.ts

### DISCUSSION THREAD

**Discussion by irririki** [DefinitelyTyped/DefinitelyTyped#59489] [Issues with a @types package] [answered]
I am using:
"jsrsasign": "^10.5.13",
"@types/jsrsasign": "^10.2.0",

In ECDSA.d.ts, the constructor is defined as 
[https://github.com/DefinitelyTyped/DefinitelyTyped/blob/173aa9174685d78871a5c035b74d856f4aef9cc8/types/jsrsasign/modules/KJUR/crypto/ECDSA.d.ts#L17](url)
`constructor(publicKey?: { curve: string; pub?: string | undefined });`

However, in ecdsa-modified-1.0.js, the params passed in the function might be prv or pub or both, as shown in line starting from 665 
[https://github.com/kjur/jsrsasign/blob/3f79b95baa34a2c35070b43d009f318d5c42ecc2/src/ecdsa-modified-1.0.js#L665](url)
```
if (params !== undefined) {
	if (params.prv !== undefined) this.setPrivateKeyHex(params.prv);
	if (params.pub !== undefined) this.setPublicKeyHex(params.pub);
 }
```

To be in line with the code in jsrasign v10.5.13, I think the constructor should be changed to:
constructor(publicKey?: { curve: string; pub?: string | undefined; prv?: string | undefined});

[a comment by an automated account is not shown]

**Comment by MysteryBlokHed** [comment, marked as answer]
I just opened #59494
